src/Jobs/ClearIndexJob.php

```diff
@@ -24,21 +24,21 @@ class ClearIndexJob extends BatchJob
     use ServiceAware;
 
     private static array $dependencies = [
-        'IndexService' => '%$' . IndexingInterface::class,
+        'indexService' => '%$' . IndexingInterface::class,
     ];
 
-    public function __construct(?string $indexName = null, ?int $batchSize = null)
+    public function __construct(?string $indexSuffix = null, ?int $batchSize = null)
     {
         parent::__construct();
 
-        if (!$indexName) {
+        if (!$indexSuffix) {
             return;
         }
 
         // Use the provided batch size, or determine batch size from our IndexConfiguration
-        $batchSize = $batchSize ?: $this->getIndexConfigurationBatchSize(null, [$indexName]);
+        $batchSize = $batchSize ?: $this->getIndexConfigurationBatchSize(null, [$indexSuffix]);
```

Comment:
i'm struggling to see how batch size gets used here.... am I missing something?

Reply:
Hah, very true... The job seems to take this in, set the number of steps based on it, but then doesn't use it at all in the actual `process()` method.

Reply:
I think this job + the `removeAllDocuments()` on the service itself needs some re-working.

Atm `removeAllDocuments()` just does a massive `while()` loop until all documents are gone. That doesn't feel great. There would be no way (for example) to add our `batch_cooldown` feature to that.